### PR TITLE
Python CDK: Pin the airbyte_protocol version

### PR DESCRIPTION
**Pull request by girarda**
## What
Pin the CDK's dependency on airbyte-protocol-models to `^0.9.0` to avoid accidental breaking changes. We should implicitly bump to a major version if/when necessary

## Can this PR be safely reverted and rolled back?
- [X] YES 💚

### CODE REVIEW

File: airbyte-cdk/python/pyproject.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ keywords = ["airbyte", "connector-development-kit", "cdk"]
 
 [tool.poetry.dependencies]
 python = "^3.9"
-airbyte-protocol-models = "*"
+airbyte-protocol-models = "~0.9"
```
Comment (erohmensing) [inline airbyte-cdk/python/pyproject.toml:27]: If you want to keep it under 1.0, but are okay with 0.10, 0.11 etc, this should be `^0.9.0`

**Review by erohmensing — APPROVED** [review]
Thanks for this! This is just preventative right? I remember hearing something about there being protocol version dependency clashes with the CDK and something, but there doesn't seem to be a later version 🤔

File: airbyte-cdk/python/pyproject.toml (hunk 1 above)
Comment (erohmensing) [reply airbyte-cdk/python/pyproject.toml]: ... or [that might be wrong specifically for 0 versions](https://python-poetry.org/docs/dependency-specification/), now i'm confused

Comment (girarda) [reply airbyte-cdk/python/pyproject.toml]: I guess we can do `>= 0.9.0, < 2.0`

Comment (erohmensing) [reply airbyte-cdk/python/pyproject.toml]: You mean <1.0? Works!

Comment (girarda) [reply airbyte-cdk/python/pyproject.toml]: yes... 7 commits for a 1-line change 🤦

**Pull request merged**